### PR TITLE
Updated Startpage integration

### PR DESCRIPTION
**Pull request by bsclifton**
- gives Brave attribution
- fixes branding mistake; StartPage (wrong) vs Startpage (correct)

Fixes https://github.com/brave/brave-browser/issues/12257

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See https://github.com/brave/brave-browser/issues/12257

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/search_engines/brave_prepopulated_engines.cc [hunk 1]
```diff
@@ -122,11 +122,11 @@ const PrepopulatedEngine qwant = {
 };
 
 const PrepopulatedEngine startpage = {
-    L"StartPage",
+    L"Startpage",
     L":sp",
     "https://www.startpage.com/favicon.ico",
     "https://www.startpage.com/do/"
-    "dsearch?query={searchTerms}&cat=web&pl=opensearch",
+    "search?q={query}&segment=startpage.brave",
```
Comment (mkarolin) [inline components/search_engines/brave_prepopulated_engines.cc:129]: Need to bump `kBraveCurrentDataVersion` at the top of the file?

Comment (bsclifton) [reply components/search_engines/brave_prepopulated_engines.cc:129]: yes- you're absolutely right! Thanks for catching 😄

Comment (bsclifton) [reply components/search_engines/brave_prepopulated_engines.cc:129]: Fixed! 😄

Comment (mkarolin) [inline components/search_engines/brave_prepopulated_engines.cc:129]: Sorry, didn't look before, but all other engines seem to use `{searchTerms}`, does it matter that we switch to `{query}` here?

Comment (bsclifton) [reply components/search_engines/brave_prepopulated_engines.cc:129]: oh wow - yes, you're 100% right 🤦‍♂️ I copy pasted from an email (busted!) 😢 One moment here...

Comment (bsclifton) [reply components/search_engines/brave_prepopulated_engines.cc]: OK fixed - thanks again 😄 This time I compiled + tested the implementation to verify it works as expected

**Review by mkarolin — APPROVED** [review]
++ LGTM

**Comment by LaurenWags** [comment]
Verified using

```
Brave	1.18.5 Chromium: 86.0.4240.111 (Official Build) nightly (x86_64)
Revision	b8c36128a06ebad76af51591bfec980224db5522-refs/branch-heads/4240@{#1290}
OS	macOS Version 10.14.6 (Build 18G3020)
```

Verified test plan from https://github.com/brave/brave-browser/issues/12257#issue-726687878.

Confirmed new profiles behaved as described when Startpage was selected as the default search engine.

Confirmed upgrade profiles behaved as described when Startpage was selected as the default search engine.

<details>
<summary>Confirmed casing on Startpage in multiple areas</summary>

`brave://welcome` page:

Casing 1

`brave://settings/search` drop down:

Casing 2

Omnibox search:

Casing 4

Right click search:

Casing 3

</details>

<details>
<summary>Confirmed search URL is configured as expected</summary>

settings

</details>

<details>
<summary>Confirmed search URL shows as expected</summary>

search URL

</details>

**Pull request merged**